File: KIWI-Backend/src/main/java/com/d205/KIWI_Backend/order/dto/OrderResponse.java
package com.d205.KIWI_Backend.order.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class OrderResponse {
    private Long orderId;  // 주문 ID
    private LocalDateTime orderDate;  // 주문 날짜
    private String status;  // 주문 상태
    private List<MenuOrderResponse> menuOrders;  // 주문한 메뉴들
    private Integer totalPrice;  // 총 가격
    private Integer kioskId;  // 키오스크 ID 추가
    private Integer orderNumber;

    @Getter
    @Setter
    @Builder
    public static class MenuOrderResponse {
        private Integer menuId;  // 메뉴 ID
        private String name;  // 메뉴 이름
        private Integer quantity;  // 수량
        private Integer price;  // 가격
        private Integer orderNumber; //번호표
    }
}

File: KIWE-Android/data/src/main/java/com/kiwe/data/model/response/OrderResponse.kt
package com.kiwe.data.model.response

import kotlinx.serialization.Serializable

@Serializable
data class OrderResponse(
    val orderId: Int = 0,
    val orderDate: String = "",
    val status: String = "",
    val menuOrders: List<OrderItemResponse> = listOf(),
    val totalPrice: Int = 0,
)

@Serializable
data class OrderItemResponse(
    val menuId: Int,
    val name: String,
    val quantity: Int,
    val price: Int,
)
